### PR TITLE
Added a wait() function

### PR DESCRIPTION
**Pull request by lehresman**
This PR is in response to #39.  This adds a `wait(cb)` function that executes the callback after there are no more pending items being sent to Rollbar.  I called the function `wait` instead of the suggested `shutdown` because it's more descriptive of the actual functionality.  The function doesn't actually shut anything down; it just executes a callback when there is nothing in flight.

### CODE REVIEW

**Comment by rokob** [comment]
I'm still adjusting to the fact that Node is single threaded because I keep thinking I am seeing race conditions everywhere. This looks good to me.

File: lib/notifier.js [hunk 1]
```diff
@@ -423,6 +429,26 @@ function addItem(item, callback) {
```
Comment (rokob) [inline lib/notifier.js]: Shouldn't we dequeue the item if we catch an exception from `buildItemData`?

File: lib/notifier.js [hunk 2]
```diff
@@ -401,16 +402,21 @@ function addItem(item, callback) {
   }
 
   try {
+    var pendingItem = {item:item, req:null};
+    pendingItems.push(pendingItem);
+
     buildItemData(item, function (err, data) {
       if (err) {
         return callback(err);
```
Comment (rokob) [inline lib/notifier.js]: We should also dequeue the item here

File: lib/notifier.js (hunk 1 above)
Comment (lehresman) [inline lib/notifier.js]: Yes, good catch!  I was assuming that buildItemData caught its own exceptions, but you're right.

**Comment by rokob** [comment]
I built a system like this before that was multithreaded and I just used an integer so I could do atomic inc/dec operations on it for tracking pending requests. Theoretically, we could have pendingItems just be an integer, the `push` could be replaced with a `++` and the dequeue we could still call a function which basically just does a `--` and then calls the callback if we have one.

I don't think we should change this because having the actual items in the queue is useful for debugging that we are matching our adds/removes correctly. But it may be something to think about.

**Comment by lehresman** [comment]
Yeah, I considered that.  I built it this way for debugging, but also in case we want to expose the pending items at some point.  Figured it was harmless since it's essentially just a pointer.

**Pull request merged**